cyprecice/cyprecice.pyx

```diff
@@ -1211,6 +1211,244 @@ cdef class Interface:
         self.thisptr.readScalarData (data_id, vertex_id, _value)
         return _value
 
+    def write_block_vector_gradient_data (self, data_id, vertex_ids, gradientValues):
+        """
+        Writes vector gradient data given as block. This function writes values of specified vertices to a dataID.
+        Values are provided as a block of continuous memory. Values are stored in a numpy array [N x D] where N = number
+        of vertices and D = number of gradient components.
+
+        Parameters
+        ----------
+        data_id : int
+            Data ID to write to.
+        vertex_ids : array_like
+            Indices of the vertices.
+        gradientValues : array_like
+             Gradient values differentiated in the spacial direction (dx, dy) for 2D space, (dx, dy, dz) for 3D space
+
+        Notes
+        -----
+        Previous calls:
+            Count of available elements at values matches the configured dimension
+            Count of available elements at vertex_ids matches the given size
+            Initialize() has been called
+            Data with dataID has attribute hasGradient = true
+            
+        Examples
+        --------
+        Write block gradient vector data for a 2D problem with 2 vertices:
+        >>> data_id = 1
+        >>> vertex_ids = [1, 2]
+        >>> gradientValues = np.array([[v1x_dx, v1y_dx,v1x_dy, v1y_dy], [v2x_dx, v2y_dx,v2x_dy, v2y_dy]])
+        >>> interface.write_block_vector_data(data_id, vertex_ids, gradientValues)
+
+        Write block vector data for a 3D (D=3) problem with 2 (N=2) vertices:
+        >>> data_id = 1
+        >>> vertex_ids = [1, 2]
+        >>> gradientValues = np.array([[v1x_dx, v1y_dx, v1z_dx, v1x_dy, v1y_dy, v1z_dy, v1x_dz, v1y_dz, v1z_dz], [v2x_dx, v2y_dx, v2z_dx, v2x_dy, v2y_dy, v2z_dy, v2x_dz, v2y_dz, v2z_dz]])
+        >>> interface.write_block_vector_data(data_id, vertex_ids, gradientValues)
+        """
+        check_array_like(vertex_ids, "vertex_ids", "write_block_vector_gradient_data")
+        check_array_like(gradientValues, "gradientValues", "write_block_vector_gradient_data")
+
+        if not isinstance(gradientValues, np.ndarray):
+            gradientValues = np.asarray(gradientValues)
+
+        if len(gradientValues) > 0:
+            size, dimensions = gradientValues.shape
+            assert dimensions == self.get_dimensions() * self.get_dimensions(), "Dimensions of vector data in write_block_vector_gradient_data does not match with dimensions in problem definition. Provided dimensions: {}, expected dimensions: {}".format(dimensions, self.get_dimensions() *  self.get_dimensions())
```

Comment:
Is this assertion really correct? From the example you have written above, the dimensions you get from ´
`gradientValues.shape` would still be 2 or 3 and 4 or 9